.github/workflows/ci-ptf-tests.yml

```diff
@@ -0,0 +1,88 @@
+name: "P4SDE CI Build pipeline"
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+      
+concurrency:
+  # if workflow for PR or push is already running stop it, and start new one
+  group: build_p4sde_ci-${{ github.ref }}
+  cancel-in-progress: true
+
+# Envs for infrap4d and dpdk libs
+env:
+    SDE: ${{ github.workspace }}
+    P4C_DIR: ${{ github.workspace }}/p4c
+    SDE_INSTALL: ${{ github.workspace }}/install
+    IPDK_RECIPE: ${{ github.workspace }}/ipdk.recipe
+    DEPEND_INSTALL: ${{ github.workspace }}/install
+
+jobs:
+  build_p4dpdk_ubuntu:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: 'Checkout DPDK-target'
```

Comment:
Can you please add ccache as a step before? Ccache is a compiler cache which can speed up compilation.
https://ccache.dev/

You just need to add this step:
https://github.com/p4lang/p4c/blob/main/.github/workflows/ci-validation.yml#L29